File: src/main/java/co/morillas/controller/MainController.java
package co.morillas.controller;

import co.morillas.core.domain.Cart;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.*;

import java.util.ArrayList;
import java.util.List;

@RestController
public class MainController {

    @GetMapping("/recipes")
    public List getAllRecipes() {
        return new ArrayList();
    }

    @GetMapping("/carts/{id}")
    public Cart getCart(@PathVariable Long id) {
        return new Cart();
    }

    @PostMapping("/carts/{id}/add_recipe")
    public Cart addRecipeToCart(@PathVariable Long id, @Valid @RequestBody AddRecipeRequest request) {
        return new Cart();
    }

    @DeleteMapping("/carts/{cartId}/recipes/{recipeId}")
    public void removeRecipeFromCart(@PathVariable Long cartId, @PathVariable Long recipeId) {

    }
} 

File: src/main/kotlin/co/morillas/controller/MainController.kt
package co.morillas.controller

import co.morillas.core.service.CartService
import co.morillas.core.service.RecipeService
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.*

@RestController
class MainController(
    private val cartService: CartService,
    private val recipeService: RecipeService,
) {


    @GetMapping("/recipes")
    fun getAllRecipes(): List<RecipeResponse> {
        return recipeService.getRecipes()
    }

    @GetMapping("/carts/{id}")
    fun getCart(@PathVariable id: Long): CartResponse {
        return cartService.getCart(id)
    }

    @PostMapping("/carts/{id}/add_recipe")
    fun addRecipeToCart(@PathVariable id: Long, @Valid @RequestBody request: AddRecipeRequest): CartResponse {
        return cartService.addRecipe(id, request.recipeId)
    }

    @DeleteMapping("/carts/{cartId}/recipes/{recipeId}")
    fun removeRecipeFromCart(@PathVariable cartId: Long, @PathVariable recipeId: Long): CartResponse {
        return cartService.removeRecipe(cartId, recipeId)
    }
} 